### PR TITLE
[API]Seeing shipping and billing addresses on order summary page

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A
| --------------- | -----
| Branch?         | master 
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php [hunk 1]
```diff
@@ -615,6 +617,77 @@ public function iShouldNotSeeTheThankYouPage(): void
         Assert::same($this->client->getResponse()->getStatusCode(), 400);
     }
 
+    /**
+     * @Then address to :fullName should be used for both shipping and billing of my order
+     */
+    public function iShouldSeeThisShippingAddressAsShippingAndBillingAddress($fullName): void
```
Comment (AdamKasp) [inline src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php:623]:
```suggestion
    public function iShouldSeeThisShippingAddressAsShippingAndBillingAddress(string $fullName): void
```

File: src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php [hunk 2]
```diff
@@ -615,6 +617,77 @@ public function iShouldNotSeeTheThankYouPage(): void
...
+    {
+        $this->iShouldSeeThisShippingAddressAsShippingAddress($fullName);
+        $this->iShouldSeeThisBillingAddressAsBillingAddress($fullName);
+    }
+
+    /**
+     * @Then my order's shipping address should be to :fullName
+     */
+    public function iShouldSeeThisShippingAddressAsShippingAddress($fullName): void
```
Comment (AdamKasp) [inline src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php:632]:
```suggestion
    public function iShouldSeeThisShippingAddressAsShippingAddress(string $fullName): void
```

**Review by AdamKasp — APPROVED** [review]

File: src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php [hunk 3]
```diff
@@ -615,6 +617,77 @@ public function iShouldNotSeeTheThankYouPage(): void
...
+    }
+
+    /**
+     * @Then my order's shipping address should be to :fullName
+     */
+    public function iShouldSeeThisShippingAddressAsShippingAddress(string $fullName): void
+    {
+        /** @var Response $response */
+        $response = $this->client->getResponse();
+        $name = explode(" ",$fullName);
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php:636]:
```suggestion
        $name = explode(' ', $fullName);
```

File: src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php [hunk 4]
```diff
@@ -615,6 +617,77 @@ public function iShouldNotSeeTheThankYouPage(): void
...
+
+        Assert::same($this->responseChecker->getResponseContent($response)['shippingAddress']['firstName'],
+            $name[0]
+        );
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php:640]:
```suggestion
        Assert::same(
            $this->responseChecker->getResponseContent($response)['shippingAddress']['firstName'],
            $name[0]
        );
```

File: src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php [hunk 5]
```diff
@@ -615,6 +617,77 @@ public function iShouldNotSeeTheThankYouPage(): void
...
+        $response = $this->client->getResponse();
+        $name = explode(" ",$fullName);
+
+        Assert::same($this->responseChecker->getResponseContent($response)['billingAddress']['firstName'],
+            $name[0]
+        );
+        Assert::same(
+            $this->responseChecker->getResponseContent($response)['billingAddress']['lastName'],
+            $name[1]
+        );
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php:662]: As it looks almost the same as the method above, you could extract a private method for this check with an argument `addressType` or sth like that

**Review by GSadee — APPROVED** [review]

**Review by Tomanhez — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, @arti0090! :1st_place_medal:

**Pull request merged**